digdag-standards/src/main/java/io/digdag/standards/operator/ForEachOperatorFactory.java

```diff
@@ -27,10 +27,13 @@
         implements OperatorFactory
 {
     private static Logger logger = LoggerFactory.getLogger(ForEachOperatorFactory.class);
+    private Limits limits;
 
     @Inject
-    public ForEachOperatorFactory()
-    { }
+    public ForEachOperatorFactory(Limits limits)
```

Comment:
How about this (get Limits through TaskCallbackApi)?:

```diff
diff --git a/digdag-core/src/main/java/io/digdag/core/agent/TaskCallbackApi.java b/digdag-core/src/main/java/io/digdag/core/agent/TaskCallbackApi.java
index 735b9a3..03cef93 100644
--- a/digdag-core/src/main/java/io/digdag/core/agent/TaskCallbackApi.java
+++ b/digdag-core/src/main/java/io/digdag/core/agent/TaskCallbackApi.java
@@ -49,4 +49,6 @@ public interface TaskCallbackApi
             int projectId,
             String workflowName)
         throws ResourceNotFoundException;
+
+    Limits getLimits(int siteId);
 }
diff --git a/digdag-spi/src/main/java/io/digdag/spi/OperatorFactory.java b/digdag-spi/src/main/java/io/digdag/spi/OperatorFactory.java
index 911dafb..bb02d62 100644
--- a/digdag-spi/src/main/java/io/digdag/spi/OperatorFactory.java
+++ b/digdag-spi/src/main/java/io/digdag/spi/OperatorFactory.java
@@ -6,5 +6,5 @@ public interface OperatorFactory
 {
     String getType();
 
-    Operator newTaskExecutor(Path workspacePath, TaskRequest request);
+    Operator newTaskExecutor(TaskContext context, TaskRequest request);
 }
diff --git a/digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java b/digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java
index 83a2d77..bd1bd53 100644
--- a/digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java
+++ b/digdag-core/src/main/java/io/digdag/core/agent/OperatorManager.java
@@ -256,7 +256,19 @@ public class OperatorManager
         if (factory == null) {
             throw new ConfigException("Unknown task type: " + type);
         }
-        Operator operator = factory.newTaskExecutor(workspacePath, mergedRequest);
+        Operator operator = factory.newTaskExecutor(new TaskContext() {
+            @Override
+            public Path getWorkspacePath()
+            {
+                return workspacePath;
+            }
+
+            @Override
+            public Limits getLimits()
+            {
+                return callback.getLimits(mergedRequest.getSiteId());
+            }
+        }, mergedRequest);
 
         return operator.run();
     }
```

TaskContext can have other methods such as `setStateParams(Config)`